text/0006-BundleContext-Install-API.md

```diff
@@ -0,0 +1,136 @@
+- Start Date: 2020-06-09
+- RFC PR: (in a subsequent commit to the PR, fill me with the PR's URL)
+- CppMicroServices Issue: (fill me with the URL of the CppMicroServices issue that necessisated this RFC, otherwise leave blank)
+
+# Add BundleContext::InstallBundles API
+
+## Summary
+
+Provide a api to initialize the bundle registry with a bundle manifest
+
+## Motivation
+
+Allows users of the system to inject bundle manifests into the registry directly. We will be able to use this mechanism to speed up initial install of very large numbers of bundles.
+
+## Detailed design
+
+### BundleContext
+
+```c++
+/** Wrapper around BundleRegistry::Install
+* 
+* @param location The absolute path to the location of the bundle whose manifest 
+*        is to be installed
+* @param manifest the manifest for the bundle. 
+*/
+vector<Bundle> BundleContext::InstallBundles(std::string location
+                                             , std::vector<AnyMap> manifest);
```

Comment:
how are statically linked bundles handled? how does someone call this API for a location containing statically linked bundles?

Reply:
There's a description in the "How do we teach this" section of the doc, and a "TODO" comment in the actual header.